### PR TITLE
feat: remove poppler-utils dependency by using pypdf for page counts

### PR DESCRIPTION
**Pull request by willstott101**
To make our containers smaller and have fewer CVEs detected - I've been removing dependencies we don't use from our unstructured deployment. poppler-utils on debian trixie for instance has some CVEs associated with it.

Once I realised this project has almost entirely moved away from pdf2image in https://github.com/Unstructured-IO/unstructured/pull/4185 I thought I might as-well contribute finishing the job.

I have made some effort to ensure this is not a regression in performance or behaviour. From limited testing I think this is faster for small PDFs (probably due to no sub-process startup) and on-par for larger PDFs (thanks to opening the file and only reading the amount we need).

This code in this PR is written by Cursor with my supervision - I am heavily relying on the test suite to validate behaviour. If you are aware of any corner cases that aren't captured by the existing suite in terms of the sorts of weird PDFs that this PR might trip up on lmk and I can try and expand it.

<a href="https://cubic.dev/pr/Unstructured-IO/unstructured/pull/4433?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: unstructured/partition/pdf_image/pypdf_utils.py [hunk 1]
```diff
@@ -2,6 +2,15 @@
...
+
+def pdf_page_count(reader: PdfReader) -> int:
+    """Return the catalog page count without flattening the page tree.
+
+    Matches poppler ``pdfinfo`` ``Pages`` (PDF catalog ``/Pages`` ``/Count``).
+    """
+    return int(reader.root_object["/Pages"]["/Count"])
 
 
 def get_page_data(fp: BinaryIO, page_number: int):
```
[the remark this answers is not in the document]
Comment (willstott101) [reply unstructured/partition/pdf_image/pypdf_utils.py:16]: I'm not interested in this without a reproducing test case. I can look for such a PDF - but I think `instead of using PdfReader.get_num_pages() (which the replaced code called)` is miss-selling it a bit. We changes one call-site that did that, but mostly replaced call-sites that did indeed rely on metadata. So this is aligning all page counting to rely on metadata, which I think is more defensible that replacing all page counting with metadata.
